src/Microsoft.DotNet.ProjectModel/WorkspaceContext.cs

```diff
@@ -445,7 +445,13 @@ private static string NormalizeProjectPath(string path)
             if (File.Exists(path) &&
                 string.Equals(Path.GetFileName(path), Project.FileName, StringComparison.OrdinalIgnoreCase))
             {
-                return Path.GetFullPath(Path.GetDirectoryName(path));
+                string directoryName = Path.GetDirectoryName(path);
```

Comment:
I'm not sure we need to test everything end to end. This same bug exists for publish, and run. Should we add tests for all of them?

My thought was to unit test this at the common place, and all the callsites don't need to be tested.